src/modules/75-cd/components/EnvironmentGroups/EnvironmentGroupDetails/EditEnvironmentGroupModal.tsx

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { FormEvent, useMemo, useRef, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import { defaultTo, isEmpty, isEqual } from 'lodash-es'
+import { Spinner } from '@blueprintjs/core'
+
+import {
+  Button,
+  ButtonVariation,
+  Container,
+  ExpandingSearchInput,
+  Heading,
+  Layout,
+  TagsPopover,
+  Text
+} from '@harness/uicore'
+import { Color, FontVariation } from '@harness/design-system'
+import { useStrings } from 'framework/strings'
+import { EnvironmentGroupResponse, EnvironmentResponse, getEnvironmentListPromise } from 'services/cd-ng'
+
+import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
+
+import { useInfiniteScroll } from '@templates-library/components/TemplateActivityLog/InfiniteScroll'
+
+import { EnvironmentSelection } from '../CreateEnvironmentGroupModal'
+
+import EmptyEnvironmentGroup from '../images/EmptyEnvironmentGroup.svg'
+
+import css from './EnvironmentGroupDetails.module.scss'
+
+export default function EditEnvironmentGroupModal({
+  selectedEnvs,
+  onEnvironmentUpdate
+}: {
+  selectedEnvs: EnvironmentGroupResponse[]
+  onEnvironmentUpdate: (envsChanged: boolean, newEnvs?: EnvironmentResponse[]) => void
+}) {
+  const { accountId, orgIdentifier, projectIdentifier } = useParams<ProjectPathProps>()
+  const [newEnvs, setNewEnvs] = useState<EnvironmentResponse[]>(selectedEnvs)
+  const [searchTerm, setSearchTerm] = useState('')
+
+  const loadMoreRef = useRef(null)
+  const pageSize = useRef(20)
+
+  const { getString } = useStrings()
+
+  const queryParams = useMemo(
+    () => ({
+      accountIdentifier: accountId,
+      orgIdentifier,
+      projectIdentifier,
+      size: pageSize.current,
+      searchTerm
+    }),
+    [accountId, orgIdentifier, projectIdentifier, searchTerm]
+  )
+
+  const {
+    items: environments,
+    error: fetchEnvironmentsError,
+    fetching: fetchingEnvironments,
+    attachRefToLastElement,
+    offsetToFetch
+  } = useInfiniteScroll({
+    getItems: options => {
+      return getEnvironmentListPromise({
+        queryParams: { ...queryParams, size: options.limit, page: options.offset }
+      })
+    },
+    limit: pageSize.current,
+    loadMoreRef,
+    searchTerm
+  })
+
+  const isFetchingEnvironmentsFirstTime = useMemo(() => {
+    return fetchingEnvironments && offsetToFetch.current === 0
+  }, [fetchingEnvironments, offsetToFetch.current])
```

Comment:
Codacy has a fix for the issue: [React Hook useMemo has a missing dependency: 'offsetToFetch'. Either include it or remove the dependency array. Mutable values like 'offsetToFetch.current' aren't valid dependencies because mutating them doesn't re-render the component.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9264786)
```suggestion
  }, [fetchingEnvironments, offsetToFetch])
```